### PR TITLE
`azurerm_site_recovery_replication_recovery_plan`: support `azure_to_azure_settings`

### LINKED ISSUE #21584: Support for zone-redundant site recovery replication recovery plans
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The current version of azurerm_site_recovery_replication_recovery_plan does not support the creation of recovery plans for VMs protected by Availability Zone to Availability Zone replication. 

The last versions of ARM Template provide therefore the property providerSpecificInput:
```
"providerSpecificInput" : [
    {
      "instanceType" : "A2A",
      "primaryZone" : "1",
      "recoveryZone" : "2"
    }
  ] 
```
https://learn.microsoft.com/en-us/azure/templates/microsoft.recoveryservices/2022-10-01/vaults/replicationrecoveryplans?pivots=deployment-language-arm-template#recoveryplana2ainput-1

Please add support for this property. Currently it's just possible to add recovery plans for zone zo zone replication via azurerm_template_deployment without the possibility to update the plan later, without destoning it first. 

### New or Affected Resource(s)/Data Source(s)

azurerm_site_recovery_replication_recovery_plan 

### Potential Terraform Configuration

```hcl
resource "azurerm_site_recovery_replication_recovery_plan" "example" {
  name                      = "example-recover-plan"
  recovery_vault_id         = azurerm_recovery_services_vault.target.id
  source_recovery_fabric_id = azurerm_site_recovery_fabric.source.id
  target_recovery_fabric_id = azurerm_site_recovery_fabric.target.id

  instanceTyp" : "A2A"
  primaryZone :  1
  recoveryZone : 2
  
  recovery_group {
    type                       = "Boot"
    replicated_protected_items = [azurerm_site_recovery_replicated_vm.test.id]
  }
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by ziyeqf**
fix #21584

test
---
```
tftest recoveryservices TestAccSiteRecoveryReplicationRecoveryPlan          
=== RUN   TestAccSiteRecoveryReplicationRecoveryPlan_basic
=== PAUSE TestAccSiteRecoveryReplicationRecoveryPlan_basic
=== RUN   TestAccSiteRecoveryReplicationRecoveryPlan_withPreActions
=== PAUSE TestAccSiteRecoveryReplicationRecoveryPlan_withPreActions
=== RUN   TestAccSiteRecoveryReplicationRecoveryPlan_withPostActions
=== PAUSE TestAccSiteRecoveryReplicationRecoveryPlan_withPostActions
=== RUN   TestAccSiteRecoveryReplicationRecoveryPlan_withZones
=== PAUSE TestAccSiteRecoveryReplicationRecoveryPlan_withZones
=== CONT  TestAccSiteRecoveryReplicationRecoveryPlan_basic
=== CONT  TestAccSiteRecoveryReplicationRecoveryPlan_withPreActions
=== CONT  TestAccSiteRecoveryReplicationRecoveryPlan_withZones
=== CONT  TestAccSiteRecoveryReplicationRecoveryPlan_withPostActions
--- PASS: TestAccSiteRecoveryReplicationRecoveryPlan_withPreActions (2968.79s)
--- PASS: TestAccSiteRecoveryReplicationRecoveryPlan_basic (3000.41s)
--- PASS: TestAccSiteRecoveryReplicationRecoveryPlan_withPostActions (3096.27s)
--- PASS: TestAccSiteRecoveryReplicationRecoveryPlan_withZones (3352.37s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices      3352.441s
```

### CODE REVIEW

**Comment by ziyeqf** [comment]
@catriona-m sure, test case added. Sorry for missing that one.
Screenshot 2023-05-06 at 11 04 58

File: internal/services/recoveryservices/site_recovery_replication_recovery_plan_resource.go [hunk 1]
```diff
@@ -250,6 +309,17 @@ func (r SiteRecoveryReplicationRecoveryPlanResource) Create() sdk.ResourceFunc {
...
+				parameters.Properties.ProviderSpecificInput = pointer.To([]replicationrecoveryplans.RecoveryPlanProviderSpecificInput{
+					replicationrecoveryplans.RecoveryPlanA2AInput{
+						PrimaryZone:              &model.A2ASettings[0].PrimaryZone,
+						RecoveryZone:             &model.A2ASettings[0].RecoveryZone,
+						PrimaryExtendedLocation:  expandEdgeZone(model.A2ASettings[0].PrimaryEdgeZone),
+						RecoveryExtendedLocation: expandEdgeZone(model.A2ASettings[0].RecoveryEdgeZone),
+					},
+				})
+			}
+
```
Comment (catriona-m) [inline internal/services/recoveryservices/site_recovery_replication_recovery_plan_resource.go:322]: could we move this into an expand function?

File: internal/services/recoveryservices/site_recovery_replication_recovery_plan_resource.go [hunk 2]
```diff
@@ -300,10 +370,22 @@ func (r SiteRecoveryReplicationRecoveryPlanResource) Read() sdk.ResourceFunc {
...
+						state.A2ASettings = []ReplicationRecoveryPlanA2ASpecificInputModel{
+							{
+								PrimaryZone:      pointer.From(a2a.PrimaryZone),
+								RecoveryZone:     pointer.From(a2a.RecoveryZone),
+								PrimaryEdgeZone:  flattenEdgeZone(a2a.PrimaryExtendedLocation),
+								RecoveryEdgeZone: flattenEdgeZone(a2a.RecoveryExtendedLocation),
+							},
+						}
+					}
+				}
```
Comment (catriona-m) [inline internal/services/recoveryservices/site_recovery_replication_recovery_plan_resource.go:388]: could we move this into a flatten function?

File: website/docs/r/site_recovery_replication_recovery_plan.html.markdown [hunk 3]
```diff
@@ -3,12 +3,12 @@ subcategory: "Recovery Services"
 layout: "azurerm"
 page_title: "Azure Resource Manager: azurerm_site_recovery_replication_recovery_plan"
 description: |-
-    Manages an Azure Site Recovery Plan within a Recovery Services vault.
+    Manages an Site Recovery Replication Recovery Plan within a Recovery Services vault.
```
Comment (catriona-m) [inline website/docs/r/site_recovery_replication_recovery_plan.html.markdown:6]:
```suggestion
    Manages a Site Recovery Replication Recovery Plan within a Recovery Services vault.
```

File: website/docs/r/site_recovery_replication_recovery_plan.html.markdown [hunk 4]
```diff
@@ -3,12 +3,12 @@ subcategory: "Recovery Services"
...
 ---
 
 # azurerm_site_recovery_replication_recovery_plan
 
-Manages an Azure Site Recovery Plan within a Recovery Services vault. A recovery plan gathers machines into recovery groups for the purpose of failover.
+Manages an Site Recovery Replication Recovery Plan within a Recovery Services vault. A recovery plan gathers machines into recovery groups for the purpose of failover.
```
Comment (catriona-m) [inline website/docs/r/site_recovery_replication_recovery_plan.html.markdown:11]:
```suggestion
Manages a Site Recovery Replication Recovery Plan within a Recovery Services vault. A recovery plan gathers machines into recovery groups for the purpose of failover.
```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks @ziyeqf - just spotted a couple of minor things on having a second look, but otherwise looking pretty good! Thanks!

**Review by katbyte — APPROVED** [review]
LGTM 📺

**Pull request merged**